doc/developer_guide/conf.py

```diff
@@ -89,7 +89,7 @@
 
 # General information about the project.
 project = u"PSyclone Developer Guide"
-copyright = u'2017-2023, STFC Daresbury Laboratory'
+copyright = u'2017-2024, STFC Daresbury Laboratory'
```

Comment:
I've done some tidying. I don't think there's any alternative to `exec` here but it's not accepting user input - it's simply executing one of the files in the repository. I've disabled the warning. Strictly speaking, the license is optional for every file in the repo because we have a LICENSE file at the base. We've not been as thorough with these configuration files so I suggest we leave it as is.